benchmark/Streamly/Benchmark/Data/Parser/ParserD.hs

```diff
@@ -192,6 +192,25 @@ spanByRolling :: MonadThrow m => Int -> SerialT m Int -> m ((), ())
 spanByRolling value =
     IP.parseD (PR.spanByRolling (\_ i -> i <= value `div` 2) FL.drain FL.drain)
 
+-------------------------------------------------------------------------------
+-- Nested
+-------------------------------------------------------------------------------
+
+{-# INLINE parseMany #-}
+parseMany :: MonadThrow m => Int -> SerialT m Int -> m ()
+parseMany value = IP.drain . IP.parseManyD (drainWhile (<= value + 1))
+
+{-# INLINE parseManyGroups #-}
+parseManyGroups :: MonadThrow m => Int -> SerialT m Int -> m ()
+parseManyGroups value =
+    IP.drain . IP.parseManyD (PR.groupBy (\_ i -> i <= value + 1) FL.drain)
```

Comment:
It will be better to put each element in its own group, so that we can test the performance of group creation path. Currently we are testing only a single group.

Reply:
Yeah, the goal was to only check for fusion.

Reply:
```
benchmarked Data.Parser.ParserD/o-1-space/parseMany groupBy (bound groups)
time                 5.105 ms   (5.076 ms .. 5.142 ms)
                     1.000 R²   (0.999 R² .. 1.000 R²)
mean                 5.111 ms   (5.100 ms .. 5.129 ms)
std dev              33.17 μs   (18.37 μs .. 55.00 μs)

benchmarked Data.Parser.ParserD/o-1-space/parseMany groupRollingBy (bound groups)
time                 5.143 ms   (5.109 ms .. 5.186 ms)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 5.136 ms   (5.114 ms .. 5.153 ms)
std dev              45.18 μs   (31.18 μs .. 69.42 μs)

benchmarked Data.Parser.ParserD/o-1-space/parseMany groupBy (1 group)
time                 51.62 μs   (51.36 μs .. 51.85 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 52.25 μs   (51.89 μs .. 52.80 μs)
std dev              2.158 μs   (1.647 μs .. 2.846 μs)
variance introduced by outliers: 38% (moderately inflated)

benchmarked Data.Parser.ParserD/o-1-space/parseMany groupRollingBy (1 group)
time                 51.44 μs   (51.17 μs .. 51.71 μs)
                     1.000 R²   (1.000 R² .. 1.000 R²)
mean                 51.57 μs   (51.35 μs .. 51.81 μs)
std dev              1.189 μs   (1.017 μs .. 1.478 μs)
variance introduced by outliers: 16% (moderately inflated)
```

Reply:
Is this the expected result?

Reply:
its the cost overhead of group processing, unless we see some obvious inefficiency in group processing.